pkgs/by-name/wa/walksnail-osd-tool/package.nix

```
@@ -0,0 +1,81 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, makeWrapper
+, wrapGAppsHook3
+, makeDesktopItem
+, copyDesktopItems
+, rustPlatform
+, cargo
+, cmake
+, pkg-config
+, libGL
+, openssl
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "walksnail-osd-tool";
+  version = "0.3.0"; # don't forget to update the make-build-reproducible.patch
+
+  src = fetchFromGitHub {
+    owner = "avsaase";
+    repo = "walksnail-osd-tool";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-xCrshFRsM4qUF4TffZiriNaplkjif/LeFOwLxoqtwsY=";
+  };
+
+  patches = [
+    ./make-build-reproducible.patch
+  ];
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "ffprobe-0.3.3" = "sha256-T6ht3ZS68Hsfw+t9uGO+ZveZJtvEH3nCd+N3vh8g5HM=";
+      "poll-promise-0.2.0" = "sha256-IrV0asFTu2P/FW60ft0PphFzfmkR08M/YBwUXuGVRXk=";
+    };
+  };
+
+  nativeBuildInputs = [
+    cargo
+    cmake
+    copyDesktopItems
+    openssl.dev
+    pkg-config
+    rustPlatform.cargoSetupHook
+    wrapGAppsHook3
+  ];
+  buildInputs = [
+    openssl
+  ];
+  env = {
+    OPENSSL_NO_VENDOR = 1;
+    VERGEN_CARGO_TARGET_TRIPLE = "x86_64-unknown-linux-gnu";
+    VERGEN_RUSTC_SEMVER = "nixos-unstable-latest";
+    GIT_VERSION = "v${version}";
+    GIT_COMMIT_HASH = "v${version}";
+  };
+
+  postFixup = ''
+    patchelf $out/bin/.${pname}-wrapped \
+              --add-rpath ${lib.makeLibraryPath [ libGL ]}
```

```suggestion
    patchelf $out/bin/.walksnail-osd-tool-wrapped \
      --add-rpath ${lib.makeLibraryPath [ libGL ]}
```
Also doing that preFixup if possible, would allow us to use the unwrapped name.